### ISSUE TITLE
Please make WSA open source

### ISSUE THREAD

**Issue by berkekbgz** [microsoft/WSA#539]
### Is your feature request related to a problem? Please describe

Discontinuation of the WSA

### Describe the solution you'd like

WSA is among the best features of Windows 11 and killing it completely is sad news for both developers and users. Instead of killing it altogether, it would be more beneficial to open source the project. Open-source community could potentially take over the development and maintenance efforts and this valuable feature continues to be supported.

### Describe alternatives you've considered

--

### Please specify the version of Windows Subsystem for Android

--

**Comment by gr9** [comment]
I was just thinking the same thing. Hear hear.

**Comment by XantreDev** [comment]
I hope they will open source it

**Comment by lurenjia534** [comment]
Me too, I hope they can be open sourced. Otherwise Windows 11 has almost no value

**Comment by MeowLove** [comment]
Yes, my suggestion is that Microsoft should spend as much time as possible in the one-year period from March 5, 2024 to March 5, 2025. Turn WSA (Windows Subsystem for Android) into community support. Microsoft will only focus on the update support of WSL2 or WSL3 (future).

For ordinary users, they can run a large number of Android applications on Windows;
For developers, it is very convenient for us to develop applications on windows through wsa.

After communization, for Microsoft's business strategy, only AOSP can be integrated without any application store to avoid legal risks with a single partner (such as Amazon). It is a very necessary function for PC to support mobile phone applications. There is no doubt about it.

Secondary modification and addition:
Regarding how to install applications only with aosp, Microsoft can build an interface through WSA, just like the WSA enhanced toolbox developed by a third party at present, and install apk applications selected in explorer through adb.

**Comment by NiffirgkcaJ** [comment]
I second this. Since, all the efforts that these developers have put into this wonderful project will be wasted like it was nothing.

**Comment by Slion** [comment]
It can be really costly to open source such large projects. Though in WSA case a lot of it is already open source so maybe it's not that bad. To be fair I don't really care so much if it is fully open source or not. I would just hate to see it definitely discountinued and unmaintained. That project had so much potential. The integration with Windows is already really good.

Actually made that post using [Fulguris](https://github.com/Slion/Fulguris) on Windows 11 thanks to WSA.

**Comment by tomrus88** [comment]
It should definitely be open sourced. If MS don't want to keep developing it themselves, at least let community do so...

**Comment by laracroftonline** [comment]
I agree with all of the above remarks.

**Comment by lemisky** [comment]
+1

**Comment by mempler** [comment]
I mean, its very unlikely that microsoft will decide to open source it. How about we finish up implementing wine and ditch windows altogether?

That way we're no longer bound to use windows. Linux has native android support thanks to waydroid.


Jokes aside, we should prob just reimplement WSA ourself, WSL2 also isn't open source either; only some components and WSA is effectively a fork of WSL2 with some modifications.

But we have the base operating system of wsl2 and we also know what it runs on which has a public API, so reimplementing it from scratch isn't impossible. Just takes a long time.

**Comment by 7gxycn08** [comment]
If they don't release the source code eventually the Russians will.

**Comment by rom4ster** [comment]
> It can be really costly to open source such large projects. Though in WSA case a lot of it is already open source so maybe it's not that bad. To be fair I don't really care so much if it is fully open source or not. I would just hate to see it definitely discountinued and unmaintained. That project had so much potential. The integration with Windows is already really good.
> 
> Actually made that post using [Fulguris](https://github.com/Slion/Fulguris) on Windows 11 thanks to WSA.

Ez fix, open source then public archive. Then the forks will come.

**Comment by lurenjia534** [comment]
> I mean, its very unlikely that microsoft will decide to open source it. How about we finish up implementing wine and ditch windows altogether?
> 
> That way we're no longer bound to use windows. Linux has native android support thanks to waydroid.
> 
> Jokes aside, we should prob just reimplement WSA ourself, WSL2 also isn't open source either; only some components and WSA is effectively a fork of WSL2 with some modifications.
> 
> But we have the base operating system of wsl2 and we also know what it runs on which has a public API, so reimplementing it from scratch isn't impossible. Just takes a long time.

You mean; implement WayDroid from WSL2?

**Comment by mempler** [comment]
> > I mean, its very unlikely that microsoft will decide to open source it. How about we finish up implementing wine and ditch windows altogether?
> > That way we're no longer bound to use windows. Linux has native android support thanks to waydroid.
> > Jokes aside, we should prob just reimplement WSA ourself, WSL2 also isn't open source either; only some components and WSA is effectively a fork of WSL2 with some modifications.
> > But we have the base operating system of wsl2 and we also know what it runs on which has a public API, so reimplementing it from scratch isn't impossible. Just takes a long time.
> 
> You mean; implement WayDroid from WSL2?

nope, ditching Windows altogether

**Comment by mempler** [comment]
But now that i think about it, we can actually use waydroid to implement wsa on wsl2. Since wsl2 should already use a waydroid compatible kernel due to wsa

Unless microsoft made a complete alternative to wsl2, which i doubt

**Comment by ElsAr4e** [comment]
Probably useful for this project: https://github.com/MustardChef/WSABuilds/issues/330

**Comment by miles5600** [comment]
upvote this in the feedback hub to get Microsoft's attention

https://aka.ms/AAqk69c

**Comment by GitHub-Creeper** [comment]
Yup and we can switch to open-source and switch to free license. Reduce cost, easy to maintain with community interest.
